native_client/deepspeech.cc

```diff
@@ -429,7 +478,37 @@ ModelState::decode(vector<float>& logits)
     inputs.data(), n_frames, num_classes, *alphabet, beam_width,
     cutoff_prob, cutoff_top_n, scorer);
 
-  return strdup(alphabet->LabelsToString(out[0].tokens).c_str());
+  return out;
+}
+
+Metadata* ModelState::decode_metadata(vector<float>& logits) 
+{
+  vector<Output> out = decode_raw(logits);
+
+  Metadata* metadata = (Metadata*)malloc(sizeof (Metadata));
```

Comment:
Shouldn't we use `new` /  `delete` instead of `malloc` / `free` ?

Reply:
Why? (The entire C/C++ API is a bit of a random mix of C/C++)

Reply:
Well everywhere else we did make the effort of being consistent with delete / new in the API

Reply:
The Metadata object is returned from the public API, which is intended to be C only.

Reply:
@dabinat And that's why it's `struct`, but check how we deal with `ModelState` and `StreamingState`, in `DS_CreateModel()`